pkg/datastore/applicationstore.go

```diff
@@ -146,7 +147,7 @@ func (s *applicationStore) UpdateApplication(ctx context.Context, id string, upd
 	return s.ds.Update(ctx, ApplicationModelKind, id, applicationFactory, func(e interface{}) error {
 		a := e.(*model.Application)
 		if a.Deleted {
-			return errors.New("unable to update a deleted application")
+			return fmt.Errorf("can't update a deleted application: %w", ErrInvalidArgument)
```

Comment:
How about other handers such as `EnableApplication`, `DisableApplication`. Should we return the same error in such cases? 🤔

Reply:
I'm gonna say yes. All operations to deleted apps are obviously invalid.

Reply:
Okay, then plz keep the consistent behavior for those cases 👍 You can address it by this PR or by separated PR, it's up to you (since the name of this PR is for update handler, but I feel the change is not much) 🙌

Reply:
Okay, let me address in this PR

Reply:
Good catch. Done addressed.